src/libraries/System.Net.NameResolution/src/System/Net/Dns.cs

```diff
@@ -695,7 +704,7 @@ private static Task<TResult> RunAsync<TResult>(Func<object, long, TResult> func,
                         {
                             ((ICollection<KeyValuePair<object, Task>>)s_tasks).Remove(new KeyValuePair<object, Task>(key!, task));
                             // Since it was canceled, func(..) had not executed and call AfterResolution it needs to be called here.
-                            NameResolutionTelemetry.Log.AfterResolution(key!, startingTimestamp, new OperationCanceledException());
+                            NameResolutionTelemetry.Log.AfterResolution(key!, activity, new OperationCanceledException());
```

Comment:
Stopping the activity will set the captured `previousActivity` as the current. This will not accidentally mess anything up? I'm asking because we're starting/stopping the activities here in different tasks / threads.

Reply:
Since it's a different (child) task, it should not alter the caller `ExecutionContext`.